src/calendar/event.js

```diff
@@ -4,35 +4,22 @@ export default class Event {
 	 * @param {object} json
 	 */
 	constructor(json) {
-		// sort out link URL
-		let link = ''
-		if (json.signup_options.embed.enabled == 1) {
-			// if embed signup enabled, give them the ticket URL
-			link = json.signup_options.tickets.url
-		} else if (json.signup_options.signup_enabled == 0) {
-			// if signup is disabled, even though embed signup disabled we provide URL (they can't sign up!)
-			link = CS.detectURLScheme() + CS.url + '/events/' + json.identifier
-		}
-
 		this.allDay =
 			json.datetime_start.slice(-8) == '00:00:00' && json.datetime_end.slice(-8) == '23:59:59'
-		this.brandEmblem = json.brand.emblem
-		this.category = json.category != null ? json.category.name : null
+		this.categoryId = json.category_id
 		this.description = json.description
 		this.end = dayjs(json.datetime_end)
 		this.image =
```

Comment:
If we're passing responsibility to the user to fallback to the brand image, would it be better to represent no data with null rather than an empty string, or are we just relying on JS being a bit wafty and doing if (!event.image)?

Reply:
We're relying on js being wafty really. We're generally using .length for string parameters as well. I'm also following the pattern already in the other objects.